contrib/hyperscan/matching/input_matchers/test/hyperscan_speed_test.cc

```diff
@@ -0,0 +1,68 @@
+// Note: this should be run with --compilation_mode=opt, and would benefit from
```

Comment:
Can you post the results of this benchmark as well?

Seems like you posted the other one in the issue: https://github.com/envoyproxy/envoy/issues/19472

Reply:
Ok, I have just benchmarked in the same environment. The difference between `hyperscan_speed_test` and `matcher_speed_test` is that the former compares the regex engine and the latter compares the compiled matcher.

```
-------------------------------------------------------
Benchmark             Time             CPU   Iterations
-------------------------------------------------------
BM_RE2              345 ns          345 ns      1962212
BM_Hyperscan        204 ns          204 ns      3531100
```

Reply:
+1 -- do you need to have only one match going at a time?

Reply:
This conversation has been resolved. The matcher takes benefit from thread local storage and removed mutex, so benchmark of a single thread is the same with the situation of multithreading.